docs/REVIEWING.md

```diff
@@ -73,6 +73,17 @@ Documentation is incredibly important to Vector; it is a feature and
 differentiator for Vector. Pull requests should not be merged without adequate
 documentation, nor should they be merged with "TODOs" opened for documentation.
 
+Ideally all modules should have module level documentation. Module level
+documentation can be omitted for modules where the purpose is obvious. For
+example the with the sinks there is typically a standard number of modules
+included (service.rs, sink.rs, request_builder.rs), these modules don't need
+documentation. All `pub` and `pub(crate)` functions, structs and macros must
```

Comment:
IIRC there's a lint for this that we could enforce this with. I'd like to enforce these programmatically as much as possible, even if we need to add a pile of TODOs and allows to existing code.

Reply:
You can add `#![deny(missing_docs)]` to the top of a module to enforce this.

Reply:
I think it could be a while before we are in a position to turn on that lint, but it is definitely something I would like to aim towards.